QTGUI: Incorporate new GUI controls

ghostop14 opened the pull request:
These updates expand the user interface capabilities of
GNU Radio.  This PR includes all of the controls more fully
documented here: https://github.com/ghostop14/gr-guiextra

ghostop14 commented:
I fully expect a good bit of conversation on these.  Some of the yml's such as that for the digital number sink could use a YML "expert" review.  When creating the yml, I tweaked it until the generated python was what I wanted but I suspect it could be more "elegantly" done, especially for the controls that support both message and variable updates.

smunaut commented:
Huh ... you're missing the actual files for all of this.

ghostop14 commented:
@smunaut [face palm] try now.  The git add had failed and I missed it.  There's test examples for each to exercise them.  I'm expecting feedback and updates on these so appreciate the review, especially any suggestions on polishing up the yml files.

devnulling commented on gr-qtgui/python/qtgui/azelplot.py:
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015 Free Software Foundation, Inc.
```
Please change the Copyright year to be 2020 for all new files

ghostop14 replied:
All set.  Just pushed the change.

ghostop14 commented:
Yep, easy fix.  Just let me know what works best.

devnulling commented on gr-qtgui/examples/show_controls.grc:
```diff
@@ -0,0 +1,315 @@
+    affinity: ''
+    alias: ''
+    comment: ''
+    file: /opt/sdr/gnuradio/Flowgraphs/Production/noaa19.jpeg
```
Can we put this or another example image in the `apps/` or `examples/` folder and then use a subprocess hack like here https://github.com/gnuradio/gnuradio/pull/3105 to generate the path for the installed file?

devnulling commented on gr-qtgui/examples/test_graphicitem_overlay.grc:
```diff
@@ -0,0 +1,160 @@
+    listDelay: '1.0'
+    maxoutbuf: '0'
+    minoutbuf: '0'
+    overlayList: '[{''filename'':''/opt/sdr/gnuradio/gr-guiextra38/examples/exampleoverlay.png'',''x'':10,''y'':0,''scalefactor'':1},{''filename'':''/opt/sdr/gnuradio/gr-guiextra38/examples/exampleoverlay.png'',''x'':20,''y'':20,''scalefactor'':2},{''filename'':''/opt/sdr/gnuradio/gr-guiextra38/examples/exampleoverlay.png'',''x'':30,''y'':40,''scalefactor'':2},{''filename'':''/opt/sdr/gnuradio/gr ...
```
Another hard coded path to make dynamic

devnulling commented on gr-qtgui/examples/test_msgcheckbox_str.grc:
```diff
@@ -0,0 +1,163 @@
+    affinity: ''
+    alias: ''
+    comment: ''
+    file: /opt/sdr/gnuradio/Flowgraphs/Production/noaa19.jpeg
```
Another path fixup

ghostop14 commented:
I like that subprocess trick.  I’ll work it into the examples to take care of the path issues.

ghostop14 commented:
@devnulling I got the trick incorporated and pushed.  I had to make some msg checkbox changes to allow variables to work too and I dropped in the image files, and tweaked the examples/CMakeLists.txt to actually install the examples and test images.

devnulling commented on gr-qtgui/python/qtgui/azelplot.py:
```diff
@@ -0,0 +1,123 @@
+            convertedElevation = 90.0 - elevation
+            # Note: +azimuth for the plot is measured counter-clockwise, so need to reverse it.
+            self.reddot = self.axes.plot(-azimuth * math.pi/180.0,convertedElevation,self.dotColor, markersize=8)
+            # self.reddot = self.axes.plot(-azimuth * math.pi/180.0,convertedElevation,'ro')
```
is this line needed?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+
+#from PyQt5.QtWidgets import QApplication, QWidget,  QVBoxLayout, QHBoxLayout, QGridLayout,  QSpinBox,  QSlider
+#from PyQt5.QtCore import QSize, pyqtSignal,  QPoint,  pyqtSlot,  pyqtProperty,  Qt
+#from PyQt5.QtGui import QPainter, QPalette,  QFont,  QFontMetricsF,  QPen,  QPolygon
```
Are these 3 lines needed? looks like they should be removed.

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+# First Qt and 2nd Qt are different.  You'll get errors if they're both not available, hence the import-as to avoid name collisions
+from PyQt5 import Qt, QtGui
+from PyQt5.QtCore import Qt as Qtc
+from PyQt5.QtCore import pyqtSignal,  QPoint,  pyqtSlot,  pyqtProperty, QRect
```
some double spaces here and in the imports lines below.

pylint will catch these

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+#from PyQt5.QtCore import QSize, pyqtSignal,  QPoint,  pyqtSlot,  pyqtProperty,  Qt
+#from PyQt5.QtGui import QPainter, QPalette,  QFont,  QFontMetricsF,  QPen,  QPolygon
+
+# First Qt and 2nd Qt are different.  You'll get errors if they're both not available, hence the import-as to avoid name collisions
```
This line is too long (72chars +) Please make it a 2 liner comment, and also please run pylint on all your python files here and it'll detect this and other errors

devnulling replied:
Sorry, I confused the suggested git commit line length (72) here.

GR's coding guide lines is maximum 100 chars, ideally 80 if sensible.

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+            diameter = size.height()
+            rect = QRect(center_x-diameter/2, 0, diameter,diameter)
+            brush = QBrush(QColor(self.backgroundColor),Qtc.SolidPattern)
+            # print("Painting background with color " + self.backgroundColor + " dimensions " + str(size))
```
some more prints() that could be in logging

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+            brush = QBrush(QColor(self.backgroundColor),Qtc.SolidPattern)
+            # print("Painting background with color " + self.backgroundColor + " dimensions " + str(size))
+            painter.setBrush(brush)
+            # painter.fillRect(rect, brush)
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        painter.scale(scale, scale)
+        
+        font = QFont(self.font())
+        #font.setPixelSize(10)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        metrics = QFontMetricsF(font)
+        
+        painter.setFont(font)
+        # painter.setPen(QPen(Qtc.black))
```
rm?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        # Draw the full black needle first if needed
+        if (self.needleType == NeedleFull):
+            needleTailBrush = self.palette().brush(QPalette.Shadow)
+            # needleTailColor = Qtc.black
```
rm?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        
+        # Now draw the red tip (on top of the black needle)
+        needleTipBrush = self.palette().brush(QPalette.Highlight)
+        # needleTipColor = Qtc.red
```
rm?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        needleTipBrush.setColor(needleTipColor)
+        painter.setBrush(needleTipBrush)
+        
+        #painter.drawPolygon(QPolygon([QPoint(-3, -25), QPoint(0, -45), QPoint(3, -25),QPoint(0, -30), QPoint(-3, -25)]))
```
rm?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        painter.restore()
+    
+#    def sizeHint(self):
+#        return QSize(150, 150)
```
is this still needed?

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,290 @@
+        if (time.time() - self.last)>self.update_period:
+            self.last = time.time()
+            # trigger update
+            # self.emit(SIGNAL("updatePlot(int)"), 0)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+        self.lblcontrol = QLabel(lbl, self)
+        self.lblcontrol.setAlignment(Qtc.AlignCenter)
+
+        # For whatever reason, the progressbar doesn't show the number in the bar if it's vertical, only if it's horizontal
```
another line that is too long

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+        #if (len(lbl) > 0):
+        #    self.setMaximumSize(maxSize+30, maxSize+35)
+        #else:
+        #    self.setMaximumSize(maxSize, maxSize)  
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+        self.maxValue = maxValue
+        
+        self.textfont = QFont(self.font())
+        #font.setPixelSize(10)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+    def paintEvent(self, event):
+        super().paintEvent(event)
+
+        #self.painter.setPen(self.backgroundColor)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+            painter.drawText(size.width()/2-self.metrics.width(printText)/2,size.height()/2, printText)
+
+        painter.save()
+        # painter.translate(self.width()/2, self.height()/2)
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/dialgauge.py:
```diff
@@ -0,0 +1,210 @@
+        painter.save()
+        # painter.translate(self.width()/2, self.height()/2)
+        painter.translate(self.width(), 0)
+        # painter.scale(self.width()/120.0, self.width()/120.0)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+
+
+from PyQt5.QtWidgets import QFrame,  QVBoxLayout, QWidget,  QLabel
+from PyQt5.QtGui import QPainter, QPixmap,  QFont,  QFontMetrics, QBrush, QColor
```
some more double spaces here

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        #self.setFrameStyle(QFrame.Sunken)
+        #self.setContentsMargins(2,2,2,2)
+
+        #self.setStyleSheet("border: 2;border-style: inset;")
```
are these commented lines needed still?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        
+        self.readOnly = False
+                    
+        # self.setColors(Qt.black, Qt.white)
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        else:
+            textstr = teststr
+
+        #width=fm.horizontalAdvance(textstr)
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        else:
+            textstr = str(self.getFrequency())
+
+        #width=fm.horizontalAdvance(textstr)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        
+        # So we know:
+        # - the width of the text
+        # - The mouse click position relative to 0 (pos relative to string start will be size().width() - 2 - pos.x
```
this line is too long and needs to wrap

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        clickedThousands = False
+        for i in range(1, len(textstr)+1):
+            #width = fm.horizontaladvance(textstr[-i:])
+            #widthchar = fm.horizontaladvance(textstr[-i:-i+1])
```
couple more lines to rm ?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+                clickedUp = False
+                if (self.offset.y() > self.size().height()/2):
+                    if self.debugClick:
+                        print('clicked down')
```
these should probably go into the logging api

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+                        
+                    self.setFrequency(curFreq)
+                    if (self.clickCallback is not None):
+                        # print("calling click callback")
```
more logging?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        
+        painter = QPainter(self)
+        
+        #self.painter.setPen(self.backgroundColor)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        brush.setColor(self.backgroundColor)
+        brush.setStyle(Qt.SolidPattern)
+        rect = QtCore.QRect(2, 2, size.width()-4, size.height()-4)
+        #rect = QtCore.QRect(0, 0, size.width(), size.height())
```
rm?

devnulling commented on gr-qtgui/python/qtgui/digitalnumbercontrol.py:
```diff
@@ -0,0 +1,362 @@
+        else:
+            textstr = str(self.getFrequency())
+        rect = QtCore.QRect(0, 0, size.width()-4, size.height())
+        #painter.drawText(event.rect(), Qt.AlignRight + Qt.AlignVCenter,  textstr)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/distanceradar.py:
```diff
@@ -0,0 +1,109 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+from matplotlib.figure import Figure
```
so matplotlib is an optional dependency, and we should check that it is installed before using it. 

See the "Python modules" section here: https://github.com/gnuradio/greps/blob/master/grep-0001-coding-guidelines.md

So these should be wrapped with a try / except like here: https://github.com/gnuradio/gnuradio/blob/master/gnuradio-runtime/examples/volk_benchmark/volk_plot.py#L10-L15

This might be worth more discussion if we should just make matplotlib a required dep, but this should still be wrapped up.

michaelld replied:
This is worth discussing.

devnulling commented on gr-qtgui/python/qtgui/distanceradar.py:
```diff
@@ -0,0 +1,109 @@
+        self.fig = Figure(figsize=(width, height), dpi=dpi)
+        self.fig.patch.set_facecolor(self.backgroundColor)
+        # "axisbg was deprecated, use facecolor instead"
+        # self.axes = self.fig.add_subplot(111, polar=True, axisbg=self.backgroundColor)
```
rm ?

devnulling commented on gr-qtgui/python/qtgui/distanceradar.py:
```diff
@@ -0,0 +1,109 @@
+        # Radius: np.ones(100)*5
+        # ax.plot(np.linspace(0, 2*np.pi, 100), np.ones(100)*5, color='r', linestyle='-')
+        # Each of these use 100 points.  linespace creates the angles 0-2 PI with 100 points
+        # np.ones creates a 100 point array filled with 1's then multiplies that by the scalar 5
```
should all these commented lines be rm'ed?

devnulling commented on gr-qtgui/python/qtgui/distanceradar.py:
```diff
@@ -0,0 +1,109 @@
+        # http://nullege.com/codes/search/matplotlib.pyplot.Circle
+        # Params are: Cartesian coord of center, radius, etc...
+        # circle = plt.Circle((0.0, 0.0), radius, transform=self.axes.transData._b, color="red", alpha=0.4)
+        # self.filledcircle = self.axes.add_artist(circle)
```
and what about these commented lines?

devnulling commented on gr-qtgui/python/qtgui/graphicitem.py:
```diff
@@ -0,0 +1,170 @@
+        try:    
+            overlayitem = pmt.to_python(pmt.car(msg))
+            if overlayitem is None:
+                print("[GrGraphicsItem] ERROR: overlay message contains None in the 'car' portion of the message.  Please pass in a dictionary or list of dictionaries in this portion of the message.  Each dictionary should have the following keys: filename,x,y.  Use x=y=-1 to remove an overlay item.")
```
should these be changed to `GraphicsItem` instead of `GrGraphicsItem`? 

Also for this whole section, perhaps it should be using the logging api.

devnulling commented on gr-qtgui/python/qtgui/graphicoverlay.py:
```diff
@@ -0,0 +1,82 @@
+from gnuradio import gr
+import pmt
+
+# This thread just gets us out of the sync_block's init function so the messaging system and scheduler are active.
```
this line is too long and should be wrapped

devnulling commented on gr-qtgui/python/qtgui/ledindicator.py:
```diff
@@ -0,0 +1,207 @@
+#
+
+from PyQt5.QtWidgets import QFrame, QWidget, QHBoxLayout, QVBoxLayout, QLabel
+from PyQt5.QtGui import QPainter, QPixmap,  QBrush, QColor, QPen, QFontMetricsF
```
more double spaces

devnulling commented on gr-qtgui/python/qtgui/ledindicator.py:
```diff
@@ -0,0 +1,207 @@
+from PyQt5 import QtCore
+from PyQt5.QtCore import Qt as Qtc
+from PyQt5.QtCore import QSize, QPoint, QRect
+from PyQt5.QtGui import QPixmap,QPainter, QPainterPath,QLinearGradient, QRadialGradient, QConicalGradient
```
need to add a space between `QPixmap,QPainter`

devnulling replied:
also this line is too long

devnulling commented on gr-qtgui/python/qtgui/ledindicator.py:
```diff
@@ -0,0 +1,207 @@
+
+class LabeledLEDIndicator(QFrame):
+    # Positions: 1 = above, 2=below, 3=left, 4=right
+    def __init__(self, lbl='', onColor='green', offColor='red', initialState=False, maxSize=80, position=1, alignment=1, valignment=1, parent=None):
```
also too long

devnulling commented on gr-qtgui/python/qtgui/ledindicator.py:
```diff
@@ -0,0 +1,207 @@
+        QFrame.__init__(self, parent)
+
+        #super().setPixmap(QtGui.QPixmap(":/icons/led-red-on.png"))
+        #super().setScaledContents(True)
```
rm these lines?

devnulling commented on gr-qtgui/python/qtgui/ledindicator.py:
```diff
@@ -0,0 +1,207 @@
+        
+        painter = QPainter(self)
+        
+        #self.painter.setPen(self.backgroundColor)
```
should all the commented out lines here be rm'ed? not going to highlight all of the but between here and ~176 there is several

devnulling commented on gr-qtgui/python/qtgui/levelgauge.py:
```diff
@@ -0,0 +1,214 @@
+        self.numberControl = LevelGauge(barColor, backgroundColor, minValue, maxValue, maxSize, isVertical, isFloat,scaleFactor,showValue,parent)
+        
+        # if (isVertical):
+        #    self.numberControl.setAlignment(Qtc.AlignCenter)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/levelgauge.py:
```diff
@@ -0,0 +1,214 @@
+        self.lblcontrol = QLabel(lbl, self)
+        self.lblcontrol.setAlignment(Qtc.AlignCenter)
+
+        # For whatever reason, the progressbar doesn't show the number in the bar if it's vertical, only if it's horizontal
```
line too long, needs to wrap

devnulling commented on gr-qtgui/python/qtgui/levelgauge.py:
```diff
@@ -0,0 +1,214 @@
+        #if (len(lbl) > 0):
+        #    self.setMaximumSize(maxSize+30, maxSize+35)
+        #else:
+        #    self.setMaximumSize(maxSize, maxSize)  
```
rm?

devnulling commented on gr-qtgui/python/qtgui/levelgauge.py:
```diff
@@ -0,0 +1,214 @@
+
+        
+        #super().setPixmap(QtGui.QPixmap(":/icons/led-red-on.png"))
+        #super().setScaledContents(True)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/levelgauge.py:
```diff
@@ -0,0 +1,214 @@
+            
+
+class GrLevelGauge(gr.sync_block, LabeledLevelGauge):
+    def __init__(self, lbl='', barColor='blue', backgroundColor='white', fontColor='black', minValue=0, maxValue=100, maxSize=80, isVertical=True, position=1, isFloat=False,scaleFactor=1,showValue=False, parent=None):
```
these all need to wrap

devnulling commented on gr-qtgui/python/qtgui/qa_qtgui.py:
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 #
-# Copyright 2011 Free Software Foundation, Inc.
+# Copyright 2020 Free Software Foundation, Inc.
```
Please change this to `2011, 2020`

devnulling replied:
one fixup here

devnulling commented on gr-qtgui/python/qtgui/togglebutton.py:
```diff
@@ -0,0 +1,99 @@
+        if initPressed:
+            self.setChecked(True)
+            self.state = 1
+            # self.callback(self.pressReleasedDict['Pressed'])
```
rm?

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+import pmt
+
+from PyQt5.QtWidgets import QFrame, QWidget, QHBoxLayout, QVBoxLayout, QLabel
+from PyQt5.QtGui import QPainter, QPixmap,  QBrush, QColor, QPen, QFontMetricsF
```
double spaces

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+from PyQt5 import QtCore
+from PyQt5.QtCore import Qt as Qtc
+from PyQt5.QtCore import QSize, QPoint, QRect
+from PyQt5.QtGui import QPixmap,QPainter, QPainterPath
```
need a space

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+            valign = Qtc.AlignBottom
+            
+        layout.setAlignment(halign | valign)
+        # layout.setAlignment(Qtc.AlignCenter | Qtc.AlignVCenter)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+        
+        maxWidth = max( (maxSize+4),(maxSize*2 + metrics.width(lbl)) )
+        maxHeight = max( (maxSize/2+4),(maxSize/2 + metrics.height()+2) )
+        #print('Min size: ' + str(maxWidth) + " x " + str(maxHeight))
```
logging api?

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+        QFrame.__init__(self, parent)
+
+        #super().setPixmap(QtGui.QPixmap(":/icons/led-red-on.png"))
+        #super().setScaledContents(True)
```
rm?

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+
+        switchWidth = size.width()/2 - 2            
+        switchHeight = size.height()/2 - 2
+        ## rect.moveCenter(QPoint(size.width()/2,size.height()/2))
```
rm?

devnulling commented on gr-qtgui/python/qtgui/toggleswitch.py:
```diff
@@ -0,0 +1,204 @@
+        brush = QBrush()
+
+        rect = QtCore.QRect(0, 0, size.width(), size.height())
+        #painter.fillRect(rect, brush)
```
rm?

ghostop14 commented:
@devnulling I think I hit everything.  Went through all of the classes and:
 - Shortened the lines in all files
- Removed unused code that was in comments in all files
- Switched them all over to the gr logging system (no more prints)
- Fixed the init copyright
- Added doc strings to all of them

I left the exampleoverlay.png in the example list to make it easier for someone to read (the file doesn't exist anyway it's just a placeholder example).  But I'm open to better approaches.

Let me know if I missed anything!

michaelld commented on gr-qtgui/grc/qtgui_appbackground.block.yml:
```diff
@@ -0,0 +1,76 @@
+        
+        self.setStyleSheet(styleSht)
+
+documentation: |-
```
duplicate "documentation"

michaelld commented on gr-qtgui/grc/qtgui_AutoCorrelatorSink.block.yml:
```diff
@@ -0,0 +1,73 @@
+id: qtgui_AutoCorrelatorSink
+label: QT GUI Fast Auto-Correlator Sink
```
any "documentation"?

michaelld commented on gr-qtgui/grc/qtgui_azelplot.block.yml:
```diff
@@ -0,0 +1,52 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_dialcontrol.block.yml:
```diff
@@ -0,0 +1,98 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_dialgauge.block.yml:
```diff
@@ -0,0 +1,104 @@
+    
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_distanceradar.block.yml:
```diff
@@ -0,0 +1,64 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_levelgauge.block.yml:
```diff
@@ -0,0 +1,120 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_msgcheckbox.block.yml:
```diff
@@ -0,0 +1,89 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_msgpushbutton.block.yml:
```diff
@@ -0,0 +1,71 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_togglebutton.block.yml:
```diff
@@ -0,0 +1,111 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented on gr-qtgui/grc/qtgui_toggleswitch.block.yml:
```diff
@@ -0,0 +1,115 @@
+
+    The GUI hint can be used to position the widget within the application. The hint is of the form [tab_id@tab_index]: [row, col, row_span, col_span]. Both the tab specification and the grid position are optional.
+
+file_format: 1
```
no newline at EOF?

michaelld commented:
I like this addition, even if I haven't tested it. Just need to get it cleaned up a little.

ghostop14 commented:
@michaelld I think I got all your comments integrated.  Checked each yml for the EOF and added a quick doc comment to the correlator.

ghostop14 commented:
On the tree thing, I think from the blocks updates for correctiq, some of the modules use the tree to build out the categories rather than using the category tag in the yml.  I haven't dug it to see how it gets called.  For seamless integration I'll update that tonight.  I'll go through pylint tonight too.  (Just cringing at lint output).

ghostop14 commented:
@devnulling , @michaelld give these updates a check.  I think I covered everything.  Moved blocks to use tree, cleaned up all the spacing and a number of other issues along the way.  Went through and covered the major stuff from pylint (that was..... fun).

devnulling commented on gr-qtgui/python/qtgui/compass.py:
```diff
@@ -0,0 +1,289 @@
+from gnuradio import gr
+import pmt
+
+# First Qt and 2nd Qt are different.  You'll get errors if they're both not available,
```
double space here

devnulling commented on gr-qtgui/examples/test_graphicscaled.grc:
```diff
@@ -0,0 +1,127 @@
+    affinity: ''
+    alias: ''
+    comment: ''
+    file: /tmp/gnuradio_logo.png
```
Can this be changed to one of the included example pictures?

Running it out of the box will produce:

```

Generating: '/home/user/sdr/installs/gr_3165/share/gnuradio/examples/qt-gui/test_graphicscaled.py'

Executing: /usr/bin/python3.6 -u /home/user/sdr/installs/gr_3165/share/gnuradio/examples/qt-gui/test_graphicscaled.py

gr::log 2020-02-13 21:21:07,231 :ERROR: ERROR: Unable to find file /tmp/gnuradio_logo.png
```

devnulling commented on gr-qtgui/examples/test_azelplot.grc:
```diff
@@ -0,0 +1,143 @@
+connections:
+- [blocks_null_source_0, '0', blocks_throttle_0, '0']
+- [blocks_throttle_0, '0', blocks_null_sink_0, '0']
+- [gpredict_rotor_0, az_el, variable_qtgui_azelplot_0, azel]
```
Can this be changed to another example block as `gpreidct_rotor` is not included in tree.

ghostop14 commented:
I think I have everything addressed:
- All copyrights changed
- Fixed spacing on compass.py comment
- Fixed test_graphicscaled.grc to use the subprocess trick
- Removed test_azelplot.grc example.  Didn't have a native block that could produce the input dictionary values.

michaelld commented on gr-qtgui/python/qtgui/CMakeLists.txt:
```diff
@@ -15,8 +15,23 @@ configure_file(util.py.cmakein "${CMAKE_CURRENT_BINARY_DIR}/util.py" @ONLY)
+    dialgauge.py
+    toggleswitch.py
+    graphicoverlay.py
+    AutoCorrelatorSink.py
```
please no CamelCase!

ghostop14 commented:
@michaelld all addressed.  See how this push looks.

michaelld commented on gr-qtgui/grc/qtgui.tree.yml:
```diff
@@ -21,3 +21,20 @@
     - variable_qtgui_entry
     - variable_qtgui_label
     - qtgui_edit_box_msg
+    - qtgui_AutoCorrelatorSink
```
thinking you mean `qtgui_autocorrelator_sink.block`

michaelld replied:
since this is the file name now

michaelld commented:
I'm tagging this as DontMerge for the moment until we get clarity on (1) what to do about the large `earth.jpg` file (~13 MB); and (2) whether GR is OK with Python class names being CamelCase. These are @marcusmueller queries ... so, hopefully he's around & can take a quick look at this PR and make a judgement!

devnulling commented:
@michaelld looks like `earth.jpg` got downsized in the last commit, its currently ` BIN +1.83 MB gr-qtgui/examples/earth.jpg `

michaelld commented:
@devnulling That's a LOT better. I'd still like some clarity from @marcusmueller . There are 3 files here of similar size:
BIN +1.83 MB gr-qtgui/examples/earth.jpg 
BIN +1.61 KB gr-qtgui/examples/gnuradio_logo.png 
BIN +1.96 KB gr-qtgui/examples/satdishsmall.png 
Not sure we want these in the repo; maybe we can get them hosted elsewhere, like what we're discussing with the PR from DrMPEG?

michaelld commented:
After further thought and discussion, I'm OK with Python script CamelCase ... as noted we already us Qt which does this, so scripts have to do this to some degree anyway. All of the files have been named with snake_case, so I think that's all taken care of ... unless @marcusmueller says otherwise! I'm still concerned about those 3 ~1.5 MB files ...

ghostop14 commented:
Since the test flowgraphs demonstrate image capabilities, I had to include a few.  But totally open to storage location within GR or otherwise.  My only thought would be if they're not included and someone goes to run the test flowgraph, it wouldn't run since the files wouldn't be present.  We'd probably need a place to store them, then notes in the flowgraph on how/where to grab them.  I did cut the size down so hopefully it's manageable.  Thoughts?

ghostop14 commented:
The one big file I cut the size down and png'd it.  It's 216K now.  The others are all really small.

marcusmueller commented:
![earth](https://user-images.githubusercontent.com/958972/74873014-a2b4a380-535e-11ea-9dca-841344bdc54a.jpg)
<<-- this is about as bad I'd go in quality, and it's insignificantly smaller than 216 kB, so I'll roll with yours.

Generally, I'm pretty wary about including large files in tree: they *never* leave history, so "temporary" including something in-tree will weigh down `git clone` for ever.

michaelld commented:
I'm OK with a 300 kB image file. That, to me, isn't a big deal. The PNG are really small. It's just this earth.jpg that's the size issue here.

devnulling commented on gr-qtgui/examples/test_digitalnumcontrol_ro.grc:
```diff
@@ -0,0 +1,183 @@
+    gui_hint: 1,0,1,1
+    label: Freq
+    type: real
+    value: freq
```
this should be `input_freq`
